infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -303,23 +272,18 @@ private ServerContext getServerContext(Map<String, Object> opts) {
      * @param serverContext server context (server information)
      * @return completable future for discovered topology
      */
-    private CompletableFuture<CorfuInterClusterReplicationServerNode> startDiscoveryService(ServerContext serverContext) {
+    private void startDiscoveryService(ServerContext serverContext) {
 
         log.info("Start Discovery Service.");
-        CompletableFuture<CorfuInterClusterReplicationServerNode> discoveryServiceCallback = new CompletableFuture<>();
 
         this.clusterManagerAdapter = buildClusterManagerAdapter(serverContext.getPluginConfigFilePath());
 
         // Start LogReplicationDiscovery Service, responsible for
         // acquiring lock, retrieving Site Manager Info and processing this info
         // so this node is initialized as Source (sender) or Sink (receiver)
-        replicationDiscoveryService = new CorfuReplicationDiscoveryService(serverContext,
-                clusterManagerAdapter, discoveryServiceCallback);
-
-        Thread replicationDiscoveryThread = new Thread(replicationDiscoveryService, "discovery-service");
-        replicationDiscoveryThread.start();
-
-        return discoveryServiceCallback;
+        replicationDiscoveryService = new CorfuReplicationDiscoveryService(
+            serverContext, clusterManagerAdapter);
+        replicationDiscoveryService.run();
```

Comment:
bootstrapComplete is mainly required to check if we need to register for the log replication lock.  The registration is not done on subsequent topology updates.  But I have removed bootstrapLogReplicationService() since most logic is inside performRoleBasedSetup().

Here is how the exception flow goes now:
1. InterclusterReplicationServer(runs in the same thread as DiscoveryService):  catches exceptions thrown from there and exits if discovery service stops.
2. Thread in InterClusterReplicationServerNode(starts the server and listens permanently): catches exceptions and exits if there is an exception.  This is same as the current behavior, just that it has moved to InterClusterReplicationServerNode